test/mainnet-fork/wstETH.t.sol

```diff
@@ -0,0 +1,283 @@
+            "AccessControlUnauthorizedAccount(address,bytes32)",
+            address(this),
+            RELAYER
+        ));
+        mainnetController.requestWithdrawFromWstETH(1e18);
+    }
+
+    function test_requestWithdrawFromWstETH_zeroMaxAmount() external {
+        vm.prank(relayer);
+        vm.expectRevert("RateLimits/zero-maxAmount");
+        mainnetController.requestWithdrawFromWstETH(1e18);
+    }
+
+    function test_requestWithdrawFromWstETH_rateLimitsBoundary() external {
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_DEPOSIT(),
+            Ethereum.WETH
+        );
+        bytes32 requestWithdrawKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_REQUEST_WITHDRAW(),
+            Ethereum.WSTETH
+        );
+
+        vm.startPrank(Ethereum.SPARK_PROXY);
+        rateLimits.setRateLimitData(depositKey,         1_000e18, uint256(1_000e18) / 1 days);
+        rateLimits.setRateLimitData(requestWithdrawKey, 500e18,   uint256(500e18) / 1 days);
+        vm.stopPrank();
+
+        deal(Ethereum.WSTETH, address(almProxy), 500e18);
+
+        vm.startPrank(relayer);
+        vm.expectRevert("RateLimits/rate-limit-exceeded");
+        mainnetController.requestWithdrawFromWstETH(500e18 + 1);
+
+        deal(Ethereum.WETH, address(almProxy), 1_000e18);
+
+        mainnetController.depositToWstETH(1_000e18);
+        mainnetController.requestWithdrawFromWstETH(500e18);
+    }
+
+}
+
+contract MainnetControllerRequestWithdrawFromWstETHTests is MainnetControllerWstETHTestBase {
+
+    function test_requestWithdrawFromWstETH() external {
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_DEPOSIT(),
+            Ethereum.WETH
+        );
+        bytes32 requestWithdrawKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_REQUEST_WITHDRAW(),
+            Ethereum.WSTETH
+        );
+
+        vm.startPrank(Ethereum.SPARK_PROXY);
+        rateLimits.setRateLimitData(depositKey,         1_000e18, uint256(1_000e18) / 1 days);
+        rateLimits.setRateLimitData(requestWithdrawKey, 1_000e18, uint256(1_000e18) / 1 days);
+        vm.stopPrank();
+
+        deal(Ethereum.WETH, address(almProxy), 1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(depositKey), 1_000e18);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   1_000e18);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 0);
+
+        vm.prank(relayer);
+        mainnetController.depositToWstETH(1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(depositKey),         0);
+        assertEq(rateLimits.getCurrentRateLimit(requestWithdrawKey), 1_000e18);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   0);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 823.029395390731625220e18);
+
```

Comment:
Convert 500 wsteth to steth and check that it matches `statuses[0].amountOfStETH`